src/app/frontend/common/components/middleellipsis/middleellipsis_controller.js

```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @final
+ */
+export default class MiddleEllipsisController {
```

Comment:
This looks awesome, thnaks!

By the way, when we have time we can improve this even more. The idea is to compute width of the container and do binary search until the ellipsized text fits. Kind of similar to what we do for column layout. But this is for the future :)